### ISSUE TITLE
rustfmt doesn't recognize x86-interrupt calling convention

### ISSUE THREAD

**Issue by phil-opp** [rust-lang-nursery/rustfmt#1375]
rustfmt throws the following error if the source contains functions using the new `x86-interrupt` calling convention (see https://github.com/rust-lang/rust/pull/39832):

```
error: invalid ABI: expected one of [cdecl, stdcall, fastcall, vectorcall, aapcs, win64, sysv64, ptx-kernel, msp430-interrupt, Rust, C, system, rust-intrinsic, rust-call, platform-intrinsic, unadjusted], found `x86-interrupt`
  --> /….rs:73:8
   |
73 | extern "x86-interrupt" fn divide_by_zero_handler(stack_frame: &mut ExceptionStackFrame) {
   |

**Comment by nrc** [comment]
I expect this just needs a Syntex update

**Comment by japaric** [comment]
This appears to be fixed as of `0.2.8-nightly (6c9ee31 2017-10-07)`. Tested with:

``` rust
extern "x86-interrupt" fn foo() {}

struct VeryVeryLongStructName;

extern "x86-interrupt" fn bar(
    a: VeryVeryLongStructName,
    b: VeryVeryLongStructName,
    c: VeryVeryLongStructName,
    d: VeryVeryLongStructName,
) {
}
```

Got no errors
